third_party/forked/acme/acme.go

```diff
@@ -490,6 +490,39 @@ func (c *Client) WaitAuthorization(ctx context.Context, url string) (*Authorizat
 	}
 }
 
+// CheckAuthorization retrieves the current status of an authorization at the given URL,
+// without polling or waiting for it to reach a final state.
+//
+// It returns a non-nil Authorization only if its Status is StatusValid.
+// In all other cases, CheckAuthorization returns an error.
+// If the Status is StatusInvalid, the returned error is of type *AuthorizationError.
+// The error includes the current authorization state. If the state is StatusPending,
+// the error also includes a RetryAfter value to indicate when the client may retry.
+func (c *Client) CheckAuthorization(ctx context.Context, url string) (*Authorization, error) {
```

Comment:
@ThatsMrTalbot There already is a GetAuthorization function, which seems to do something very similar to this function.